lib/rubocop/cop/rails/pluck_in_where.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop identifies places where `pluck` is used in `where` query methods
+      # and can be replaced with `select`.
+      #
+      # Since `pluck` is an eager method and hits the database immediately,
+      # using `select` helps to avoid additional database queries.
+      #
+      # @example
+      #   # bad
+      #   Post.where(user_id: User.active.pluck(:id))
+      #
+      #   # good
+      #   Post.where(user_id: User.active.select(:id))
+      #
+      class PluckInWhere < Cop
+        MSG = 'Use `select` instead of `pluck` within `where` query method.'
+        WHERE_METHODS = %i[where rewhere].freeze
+
+        def_node_matcher :pluck_call?, <<~PATTERN
+          (send _ :pluck ...)
+        PATTERN
+
+        def on_send(node)
+          add_offense(node, location: :selector) if pluck_call?(node) && in_where?(node)
```

Comment:
Can you use `method?` instead of `def_node_matcher`?

```diff
diff --git a/lib/rubocop/cop/rails/pluck_in_where.rb b/lib/rubocop/cop/rails/pluck_in_where.rb
index d040fa927..873dd7d02 100644
--- a/lib/rubocop/cop/rails/pluck_in_where.rb
+++ b/lib/rubocop/cop/rails/pluck_in_where.rb
@@ -20,12 +20,8 @@ module RuboCop
         MSG = 'Use `select` instead of `pluck` within `where` query method.'
         WHERE_METHODS = %i[where rewhere].freeze

-        def_node_matcher :pluck_call?, <<~PATTERN
-          (send _ :pluck ...)
-        PATTERN
-
         def on_send(node)
-          add_offense(node, location: :selector) if pluck_call?(node) && in_where?(node)
+          add_offense(node, location: :selector) if node.method?(:pluck) && in_where?(node)
         end

         def autocorrect(node)
```